jdaviz/configs/imviz/plugins/footprints/footprints.py

```diff
@@ -23,6 +24,54 @@
 __all__ = ['Footprints']
 
 
+# These functions help us determine which polygon overlay is spatially closest
+# to a user's click on the viewer. The math is based on projecting a point
+# onto line segments and computing the nearest one across all polygons.
+#
+# This is useful when multiple regions are displayed and we want to know
+# which one the user meant to interact with.
+def closest_point_on_segment(px, py, x1, y1, x2, y2):
+    """Find the closest point on a line segment to a given point."""
+
+    dx, dy = x2 - x1, y2 - y1
+    denominator = dx ** 2 + dy ** 2
+    # Calculate t: how far along the segment the projection of the point falls
+    t = ((px - x1) * dx + (py - y1) * dy) / np.where(denominator == 0, 1, denominator)
+    t = np.where(denominator == 0, 0, t)  # t = 0 if denominator was actually 0
+    t = np.clip(t, 0, 1)
+    closest_x = x1 + t * dx
+    closest_y = y1 + t * dy
+    return closest_x, closest_y
+
+
+def find_closest_polygon_point(px, py, polygons):
+    """Return the closest point on a polygon AND its overlay label from a list of polygons."""

```

Comment:
nitpick (readability): Similar comment to above
```suggestion
    """
    Return the closest point on a polygon AND its overlay label from a list of polygons.
    
    Parameters
    -----------
        px: int
            x coordinate of the point we are checking
        py: int
            y coordinate of the point we are checking
        polygons: list[Dict[str, list[int]]]
            list of polygon point pairs to compare to the point and corresponding
            overlays
        
    Returns
    --------
        closest_overlay: tuple[int, int]
            label of the clostest overlay
        closest_point: tuple[int, int]
            x and y coordinates for the closest point
    """
```

Reply:
Thanks so much for the suggestion! I didn’t commit it directly because I noticed a couple of minor things (like type annotations and a small typo), so I made a few tweaks before shipping it. I really appreciated the nudge. It also gave me a chance to clean up a few variable names for clarity. Thanks again for pointing it out!